patterns/vpc-lattice/main.tf

```diff
@@ -0,0 +1,377 @@
+
+  vpc_id     = module.cluster_vpc.vpc_id
+  subnet_ids = module.cluster_vpc.private_subnets
+
+  eks_managed_node_groups = {
+    initial = {
+      instance_types = ["m5.large"]
+
+      min_size     = 3
+      max_size     = 10
+      desired_size = 3
+    }
+  }
+
+  tags = local.tags
+}
+
+################################################################################
+# Cluster VPC
+################################################################################
+
+module "cluster_vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 5.4"
+
+  name = local.name
+  cidr = local.cluster_vpc_cidr
+
+  azs             = local.azs
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.cluster_vpc_cidr, 4, k)]
+  public_subnets  = [for k, v in local.azs : cidrsubnet(local.cluster_vpc_cidr, 8, k + 48)]
+
+  enable_nat_gateway = true
+  single_nat_gateway = true
+
+  public_subnet_tags = {
+    "kubernetes.io/role/elb" = 1
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/role/internal-elb" = 1
+  }
+
+  tags = local.tags
+}
+
+################################################################################
+# Client VPC
+################################################################################
+
+module "client_vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 5.4"
+
+  name = local.name
+  cidr = local.client_vpc_cidr
+
+  azs             = local.azs
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.client_vpc_cidr, 4, k)]
+
+  tags = local.tags
+}
+
+################################################################################
+# EKS Addons (AWS Gateway API Controller)
+################################################################################
+
+module "addons" {
+  source  = "aws-ia/eks-blueprints-addons/aws"
+  version = "~> 1.12"
+
+  cluster_name      = module.eks.cluster_name
+  cluster_endpoint  = module.eks.cluster_endpoint
+  cluster_version   = module.eks.cluster_version
+  oidc_provider_arn = module.eks.oidc_provider_arn
+
+  enable_aws_gateway_api_controller = true
+  aws_gateway_api_controller = {
+    chart_version           = "v1.0.1"
+    create_namespace        = true
+    namespace               = "aws-application-networking-system"
+    source_policy_documents = [data.aws_iam_policy_document.gateway_api_controller.json]
+    set = [
+      {
+        name  = "clusterName"
+        value = module.eks.cluster_name
+      },
+      {
+        name  = "log.level"
+        value = "debug"
+      },
+      {
+        name  = "clusterVpcId"
+        value = module.cluster_vpc.vpc_id
+    }, ]
+
+  }
+
+  tags = local.tags
+}
+
+resource "null_resource" "wait_for_crd_registration" {
+  triggers = {
+    build_number = "$(timestamp())"
+  }
+
+  provisioner "local-exec" {
+    command = <<EOF
+      echo "Waiting for crd registration of aws api gateway controller..."
+      sleep 30
+    EOF
+  }
+
+  depends_on = [module.addons]
+}
```

Comment:
I am not sure if this will solve the problem. The issue is related to the registration of the CRDs within the K8S api. I believe even if the helm chart reports back it still can take a few seconds until the CRDs are properly registered. According to the doc wait=true is already configured implicitly: https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#wait

Reply:
That is true for the terraform resource, however we set it to false on the addon module:

https://github.com/aws-ia/terraform-aws-eks-blueprints-addon/blob/327207ad17f3069fdd0a76c14d3e07936eff4582/variables.tf#L191-L195

Reply:
Got it! I've added the variable and removed the block for the waiting condition